### DISCUSSION TITLE
Does periodic mesh necessary for periodic boundary conditions?

### DISCUSSION THREAD

**Discussion by abarun22** [idaholab/moose#25019] [Q&A General]
Dear all,
I am interested in running an RVE micro-structure model applying periodic boundary conditions and material laws defined by crystal plasticity considerations. I was thinking to create neper tessellations with different grain morphological properties and estimate the resulting behavior  based on the mesh sensitivities. I chose to go ahead with 'Tets' as this tends to capture the underlying micro-structure to a reasonable accuracy. A mesh with 'Tets' will in practice never be able to maintain periodicity along the edges/surfaces of the model, which makes it hard to apply periodic BC's. Surprisingly moose does seem to have no issues running this model (attached) and produced results, although i am not sure about its accuracy. My question being that, whether the periodic boundary conditions implemented in moose requires a strict construction of a periodic mesh or not. 

![image](https://github.com/idaholab/moose/assets/40111479/ab9eede6-523e-402e-82fc-dfc63315ecfb)

Any comments on the accuracy of the results with a non-periodic mesh is most welcome.
Kind regards,
Arun

**Comment by GiudGiud** [comment]
Hello

No the periodic boundaries wont complain if the mesh is not periodic, and it wont be able to enforce peridocity if it's not

Guillaume
